glottolog3/views.py

```diff
@@ -412,9 +413,86 @@ def add_identifier(request):
     except exc.SQLAlchemyError as e:
         DBSession.rollback()
         return { 'error': '{}'.format(e) }
+    
+    # So we can use identifier object without session
+    DBSession.expunge(identifier)
+    transaction.commit()
 
-    return {'message': 'Identifier successfully added.',
-            'identifier': '%s' % identifier}
+    return { 'message': 'Identifier successfully added.',
+             'identifier': '%s' % identifier } 
+
+@view_config(
+        route_name='glottolog.identifier',
+        request_method='DELETE', 
+        renderer='json')
+def delete_identifier(request):
+    gcode = request.json_body['glottocode']
+    lang = request.json_body['lang']
+    name = request.json_body['name']
+    type = request.json_body['type']
+    desc = request.json_body['description']
+
+    identifier_id = '{0}-{1}-{2}-{3}'.format(slug(name), slug(type), slug(desc or ''), lang)
+    languoid = DBSession.query(Language) \
+                        .filter_by(id='{0}'.format(gcode)) \
+                        .first()
+
+    try:
+        id_query = DBSession.query(Identifier) \
+                            .filter(Identifier.id == identifier_id) 
+        DBSession.query(LanguageIdentifier) \
+                 .filter(and_(LanguageIdentifier.language == languoid,
+                          LanguageIdentifier.identifier == id_query.first())) \
+                 .delete()
+        num_deleted = id_query.delete()
+        DBSession.flush()
+    except exc.SQLAlchemyError as e:
+        DBSession.rollback()
+        return { 'error': '{}'.format(e) }
+    
+    # Commit if no errors
+    transaction.commit()
+
+    return { 'message': 
```

Comment:
This is open to debate but I think we should be returning `None` with a 204. See https://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api